crates/rmf_site_editor/src/widgets/view_multi_selection.rs

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2025 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+use crate::{
+    site::{Delete, NameInSite},
+    widgets::{prelude::*, MultiEditPoseWidget},
+    Icons,
+};
+use bevy::{ecs::system::SystemParam, prelude::*};
+use bevy_egui::egui::{Button, CollapsingHeader, ImageButton, ScrollArea, Ui};
+use rmf_site_egui::*;
+use rmf_site_format::{InstanceMarker, SiteID};
+use rmf_site_picking::{Select, Selection};
+
+const INSTANCES_VIEWER_HEIGHT: f32 = 200.0;
+
+/// Add a plugin for viewing and editing a list of all levels
+#[derive(Default)]
+pub struct ViewMultiSelectionPlugin {}
+
+impl Plugin for ViewMultiSelectionPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_plugins(PropertiesTilePlugin::<ViewMultiSelection>::new());
+    }
+}
+
+#[derive(SystemParam)]
+pub struct ViewMultiSelection<'w, 's> {
+    icons: Res<'w, Icons>,
+    model_instances:
+        Query<'w, 's, (Entity, &'static NameInSite, &'static SiteID), With<InstanceMarker>>,
+    selection: Res<'w, Selection>,
+    delete: EventWriter<'w, Delete>,
+    select: EventWriter<'w, Select>,
+    multi_edit_pose_widget: MultiEditPoseWidget<'w, 's>,
+}
+
+impl<'w, 's> WidgetSystem<Tile> for ViewMultiSelection<'w, 's> {
+    fn show(_: Tile, ui: &mut Ui, state: &mut SystemState<Self>, world: &mut World) -> () {
+        let mut params = state.get_mut(world);
+        if params.selection.selected.len() < 2 {
+            return;
+        }
+
+        CollapsingHeader::new("ViewMultiSelection")
```

Comment:
Currently this is being conditionally popped in and out as its own major widget in the properties panel, separated from the existing inspector widget with several other major widgets in between. I don't think this is makes sense from a UX standpoint.

Instead of being separate from the Inspector widget, I think this should be a different mode for the inspector widget that activates when there is a multi-selection happening. I would recommend the following steps:

1. Rename this to `inspect_multi_selection`
2. Move it into the `inspector` directory
3. Remove this collapsing header
4. Change the implementation of `WidgetSystem` for `Inspector` so that when `selection.selected.len() > 1` it simply calls shows this widget instead, and doesn't show the usual inspectors at all.